snapshots/devmapper/snapshotter.go

```diff
@@ -272,7 +274,12 @@ func (s *Snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 // Remove removes thin device and snapshot metadata by key
 func (s *Snapshotter) Remove(ctx context.Context, key string) error {
 	log.G(ctx).WithField("key", key).Debug("remove")
+	span := trace.SpanFromContext(ctx)
```

Comment:
For snapshotters, does it make sense to add spans on the caller side, rather than each snapshotter implementation?

Reply:
Do you mean, reading the client side span from the request and adding sub-spans to that?

Reply:
I meant, it is not desirable to ask all snapshotters to implement a set of span adding logic. They are following Snapshotter interface.

Sorry, I now think adding spans on the caller side is not possible, but we still can have a struct that adds spans, then combine that to existing Snapshotter implementations, like below.

```go
impl := devmapper.NewSnapshotter(...)
ss := NewTracedSnapshotter(impl)
```